### ISSUE TITLE
Generation works correctly without ckpt files

### ISSUE THREAD

**Issue by chesha1** [huggingface/diffusers#1841]
If I do not place ckeckpoint files(e.g. v2-1_768-ema-pruned.ckpt) in the model folder, generation works as usual.
I used to think ckpt files save the model weights after training on LAION-5B. If there is no ckpt file, the model needs to inference from scratch and produce images with low quality. 
However, the quality of generation seems to have no changes whether or not there is a ckpt files.
So, what does ckpt file literally save? Is it only used for fine-tuning or further training and has not impact on inference?

```
model_id = "./models/stable-diffusion-2-1"
device = torch.device("cuda")

pipe = StableDiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float16)
pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config)
pipe = pipe.to(device)


prompt = "apple tree"
num_inference_steps = 50
height = 768
width = 768
num_images_per_prompt = 1
guidance_scale = 7.5

PipelineOut = pipe(prompt, num_inference_steps=num_inference_steps, guidance_scale=guidance_scale, height=height, width=width, num_images_per_prompt=num_images_per_prompt)
```

**Comment by Abhinay1997** [comment]
You are right that ckpt files hold the trained weights of the model for inference. But this is when using a pytorch model directly. HuggingFace models for diffusers and transformers use an approach where each constituent model weights are stored in separate folders and a model_index.json file is generated to load the model correctly.

For example:-
At https://huggingface.co/stabilityai/stable-diffusion-2-1/tree/main
<img width="1274" alt="Screenshot 2022-12-27 at 9 31 51 PM" src="https://user-images.githubusercontent.com/24771261/209691268-0ad365f2-2f95-4cbb-9d4a-53b825173a26.png">

The .ckpt file and the .safetensors file with the same name have the same model weights. Either of them can be used to load the model weights depending on your need. Apart from this, when using the ```.from_pretrained``` method, the model loads the weights from inside the folder. 

If you look inside the vae [folder](https://huggingface.co/stabilityai/stable-diffusion-2-1/tree/main/unet). ( VAE is one of the components of the Diffusion model), you'll find a config.json, a .bin file and a .safetensors file. The config.json is used to find right the model class, and depending on availability, either the safetensors file or the bin file is used to load the model weights.

TLDR:
.ckpt/.safetensors in the main directory -> has weights of the entire diffusion model
.bin/.safetensors in sub-directories -> has weights of the component of the diffusion model.

You can test this by deleting both the .bin and .safetensors file from the vae folder and see if you can still load the model using the .from_pretrained methos.

**Comment by chesha1** [comment]
> You are right that ckpt files hold the trained weights of the model for inference. But this is when using a pytorch model directly. HuggingFace models for diffusers and transformers use an approach where each constituent model weights are stored in separate folders and a model_index.json file is generated to load the model correctly.
> 
> For example:- At https://huggingface.co/stabilityai/stable-diffusion-2-1/tree/main <img alt="Screenshot 2022-12-27 at 9 31 51 PM" width="1274" src="https://user-images.githubusercontent.com/24771261/209691268-0ad365f2-2f95-4cbb-9d4a-53b825173a26.png">
> 
> The .ckpt file and the .safetensors file with the same name have the same model weights. Either of them can be used to load the model weights depending on your need. Apart from this, when using the `.from_pretrained` method, the model loads the weights from inside the component folder (vae, unet, safety_checker etc).
> 
> If you look inside the vae [folder](https://huggingface.co/stabilityai/stable-diffusion-2-1/tree/main/unet). ( VAE is one of the components of the Diffusion model), you'll find a config.json, a .bin file and a .safetensors file. The config.json is used to find the right model class, and depending on availability, either the safetensors file or the bin file is used to load the model weights.
> 
> TLDR: .ckpt/.safetensors in the main directory -> has weights of the entire diffusion model .bin/.safetensors in sub-directories -> has weights of the component of the diffusion model.
> 
> You can test this by deleting both the .bin and .safetensors file from the vae folder and see if you can still load the model using the `.from_pretrained` method.

Thank you for your reply, which really helps me a lot.
And I still have a quesiton, many models offer checkpoints of different types. For example, [Anything v3](https://huggingface.co/Linaqruf/anything-v3.0/tree/main) offer fp16/fp32 and pruned/non-pruned checkpoints. However, in terms of the components of the models, there is only one weight file in their folder(e.g. model.bin in ./unet). Is this because the only weight file contains all of the mentioned types?

**Comment by Abhinay1997** [comment]
No, In general there is only one dtype inside the .bin file. You can check this out in colab too: https://colab.research.google.com/drive/126GHbwDw06ZNO5CBRLORqm0ewdPKazR3?usp=sharing. I loaded the Anything v3's vae model weights here.

 However you can cast the dtype to your desired type while loading using ```from_pretrained(torch_dtype="torch.float16")```. But yes, it won't be the same as using the provided float16 checkpoint.

@patrickvonplaten , correct me if I am wrong
